commands/enrich/enrich.go

```diff
@@ -0,0 +1,304 @@
+	scanErrors = appendErrorSlice(scanErrors, indexedFileProducerErrors)
+
+	scanResults := xrutils.NewAuditResults()
+	scanResults.XrayVersion = xrayVersion
+	scanResults.ScaResults = flatResults
+
+	isxml, err := isXML(scanResults.ScaResults)
+	if err != nil {
+		return
+	}
+	if isxml {
+		if err = xrutils.NewResultsWriter(scanResults).
+			AppendVulnsToXML(); err != nil {
+			return
+		}
+	} else {
+		if err = xrutils.NewResultsWriter(scanResults).
+			AppendVulnsToJson(); err != nil {
+			return
+		}
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if err = utils.RecordSecurityCommandOutput(utils.ScanCommandSummaryResult{Results: scanResults.GetSummary(), Section: utils.Binary}); err != nil {
+		return err
+	}
+
+	// If includeVulnerabilities is false it means that context was provided, so we need to check for build violations.
+	// If user provided --fail=false, don't fail the build.
+	if enrichCmd.fail {
+		if xrutils.CheckIfFailBuild(scanResults.GetScaScansXrayResults()) {
+			return xrutils.NewFailBuildError()
+		}
+	}
+	if len(scanErrors) > 0 {
+		return errorutils.CheckErrorf(scanErrors[0].ErrorMessage)
+	}
+	log.Info("Enrich process completed successfully.")
+	return nil
+}
+
+func NewEnrichCommand() *EnrichCommand {
+	return &EnrichCommand{}
+}
+
+func (enrichCmd *EnrichCommand) CommandName() string {
+	return "xr_enrich"
+}
+
+func (enrichCmd *EnrichCommand) prepareScanTasks(fileProducer, indexedFileProducer parallel.Runner, resultsArr [][]*ScanInfo, indexedFileErrors [][]formats.SimpleJsonError, fileCollectingErrorsQueue *clientutils.ErrorsQueue, xrayVersion string) {
+	go func() {
+		defer fileProducer.Done()
+		// Iterate over file-spec groups and produce indexing tasks.
+		// When encountering an error, log and move to next group.
+		specFiles := enrichCmd.spec.Files
+		artifactHandlerFunc := enrichCmd.createIndexerHandlerFunc(indexedFileProducer, resultsArr, indexedFileErrors, xrayVersion)
+		taskHandler := getAddTaskToProducerFunc(fileProducer, artifactHandlerFunc)
+
+		err := FileForIndexing(specFiles[0], taskHandler)
+		if err != nil {
+			log.Error(err)
+			fileCollectingErrorsQueue.AddError(err)
+		}
+	}()
+}
+
+func (enrichCmd *EnrichCommand) createIndexerHandlerFunc(indexedFileProducer parallel.Runner, resultsArr [][]*ScanInfo, indexedFileErrors [][]formats.SimpleJsonError, xrayVersion string) FileContext {
+	return func(filePath string) parallel.TaskFunc {
+		return func(threadId int) (err error) {
+			// Add a new task to the second producer/consumer
+			// which will send the indexed binary to Xray and then will store the received result.
+			taskFunc := func(threadId int) (err error) {
+				fileContent, err := os.ReadFile(filePath)
+				if err != nil {
+					return err
+				}
+				params := &services.XrayGraphImportParams{
+					SBOMInput: fileContent,
+					ScanType:  services.Binary,
+				}
+				if enrichCmd.progress != nil {
+					enrichCmd.progress.SetHeadlineMsg("Scanning 🔍")
+				}
```

Comment:
```suggestion
    logMsgPrefix := clientutils.GetLogMsgPrefix(threadId, false)
	log.Info(logMsgPrefix+"Scanning file:", filePath)
	if enrichCmd.progress != nil {
		enrichCmd.progress.SetHeadlineMsg("Scanning file: " + filepath.Base(filePath) + " 🗄")
	}
```
This will set the progress for every file without changes.
Check out what we do it `scan.go` and make it similar.

Reply:
no need for it